src/cli.rs

```diff
@@ -91,6 +91,23 @@ pub struct Opts {
     #[arg(long)]
     pub verify: bool,
 
+    /// Sets which port we bind the GDB Server to.
+    ///
+    /// Note: doesn't necessarily start the GDB Server. See `--gdb-on-start` and
+    /// `--gdb-on-crash`.
+    #[arg(long, env = "PROBE_RUN_GDB_PORT")]
+    pub gdb_port: Option<String>,
```

Comment:
I wondered if this should be `Option<SocketAddr>` instead? This will give users a suitable error message if they provide an invalid address.

```console
$ cargo run -- --gdbport 127.0.0:3333
    Finished dev [unoptimized + debuginfo] target(s) in 0.02s
     Running `target/debug/krate --gdbport '127.0.0:3333'`
error: invalid value '127.0.0:3333' for '--gdbport <GDBPORT>': invalid socket address syntax

For more information, try '--help'.
```

But, since `GdbInstanceConfiguration::from_session` takes a `Option<impl Into<String>>`, I am unsure.

Reply:
That's a good idea. We can always turn the valid socket address back into a string before giving it to the GDB server.

Reply:
Also we should warn if they provide a gdb-port but fail to activate either of the GDB options (-on-start or -on-fault)